htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1697,8 +1697,18 @@ public function generate($format = 'ALL', $executiondate = '', $type = 'direct-d
 	public static function buildRumNumber($row_code_client, $row_datec, $row_drum)
 	{
 		global $langs;
+		$maxRumLength = 34 - 2; // look at iban_prefix in account.class.php
```

Comment:
All what e need for a rum is a random number.
I think for a simple random number adding so many condition and code is useless.
in v17, i replaced with

`return $pre.'-'.dol_print_date($row_datec, 'dayhourlogsmall').'-'.dol_trunc($row_drum.($row_code_client ? '-'.$row_code_client : ''), 15, 'right', 'UTF-8', 1);
`